ts/components/messages/MessageCTABar.tsx

```diff
@@ -180,10 +179,7 @@ class MessageCTABar extends React.PureComponent<Props, State> {
       none
     );
 
-    const isPaid = rptId
-      .map(RptIdFromString.encode)
-      .map(_ => paymentByRptId[_] !== undefined)
-      .getOrElse(false);
+    const isPaid = maybePaidReason.isSome();
```

Comment:
why `isPaid` gets treated differently from `amount` and `rptId` ?

Reply:
also, how is this change related to the stories in scope of this PR?

Reply:
What you mean with "treated differently".
The change is related to the story for the changes made to the MessageListItemComponent (now MessageInboxItem). The old MessageListItemComponent was receiving as prop the whole paymentsByRptId object, while the new MessageInboxItem only receive the paymentByRptId related to the specific list item (as Option because it can be undefined in the redux store). For this change it was also necessary to change the MessageCTABar.